### DISCUSSION TITLE
Annotation limitation

### DISCUSSION THREAD

**Discussion by olivernybroe** [community/community#26680] [Actions]
<p>I am trying to create a linting tool which runs as a GitHub action, however I keep running into limitations on GitHub actions.</p>
<p>So it looks like only 10 annotations are shown, but when my linting tool on a codebase, there could be like 500 errors, so this is obviously not ideal.</p>
<p>It seems like with a GitHub App you are allowed to add infinite amount of annotations, which sounds perfect, however than I have to migrate to a GitHub App, which is annoying as I want to run it with GitHub actions.</p>
<p>I also tried creating a review instead, however reviews can only add comments to lines which are changed in the PR and you cannot add a review if the changes are not in a PR. Also the review API crashes/times out if you add around 20 comments in one request.</p>
<p>I am kinda in a loss, what other solutions have I forgotten to look into? Or did I oversee something?<br>
I think the bast case would be if something like the check api could be made available for GitHub actions</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/olivernybroe">@olivernybroe</a> ,</p>
<p>Currently, GitHub Actions does not support so much annotations in a workflow run, the below is a summary of the limitation:</p>
<ul>
<li>10 warning annotations and 10 error annotations per step</li>
<li>50 annotations per job (sum of annotations from all the steps)</li>
<li>50 annotations per run (separate from the job annotations, these annotations aren’t created by users</li>
</ul>
<p>There is a similar ticket as reference: <a href="https://github.community/t/maximum-number-of-annotations-that-can-be-created-using-github-actions-logging-commands/16991">https://github.community/t5/GitHub-Actions/Maximum-number-of-annotations-that-can-be-created-using-GitHub/m-p/39085/highlight/false#M3660</a></p>

**Comment by olivernybroe** [comment]
<p>Yep and that limitation is why I cannot show the errors as annotations, as 10 annotations is a really low limit.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/olivernybroe">@olivernybroe</a> ,</p>
<p>Yeah, currenly we have no any available way to bypass the limitation.</p>
<p>If your projects really need more annotations, I recommend you directly report a feature request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a>. That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Reply by lisanna-dettwyler** [reply]
I've done so here: https://github.com/orgs/community/discussions/68471. Was very frustrated to run into this.

**Comment by Simran-B** [comment]
<p>It’s possible to output Markdown summaries now, which can be used to generate nicely formatted reports:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/7819991/181099187-0b79f501-2b50-43f1-95e2-17b508cd5135.png" class="site-icon" width="32" height="32">

      <a href="https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/" target="_blank" rel="noopener nofollow ugc" title="04:00PM - 09 May 2022">The GitHub Blog – 9 May 22</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://github.blog/wp-content/uploads/2022/05/166843725-62d183c8-91d7-4d3f-8ceb-4e3be5a08e45.png?fit=2048%2C1024" class="thumbnail" width="690" height="345"></div>

<h3><a href="https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/" target="_blank" rel="noopener nofollow ugc">Supercharging GitHub Actions with Job Summaries | The GitHub Blog</a></h3>

  <p>You can now output and group custom Markdown content on the Actions run summary page.</p>

  <p>
    <span class="label1">Est. reading time: 3 minutes</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by igiona** [comment]
@olivernybroe have file a feature request? If yes could you link it here?

**Reply by yzhaoa** [reply]
For the record, you can now exceed the 10 annotation limit using the checks API directly:

1. Refer to https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28, under output.annotations, in particular how there is a limit of 50 annotations in the create request, and how you can exceed that limit by calling update with more annotations.
2. To call this API more easily, use https://github.com/actions/github-script which allows you to call GitHub APIs with workflow permissions.

I have tested this and generated almost 1000 annotations on a commit successfully.

That said, I do recommend only generating annotations on changed files in a PR. Otherwise the PR will get polluted with a ton of unrelated files and annotations on them.
